internal/eval/eval.go

```diff
@@ -4115,7 +4115,7 @@ func evalJSONSTRLEN(args []string, store *dstore.Store) []byte {
 		jsonData := obj.Value
 
 		if utils.GetJSONFieldType(jsonData) != utils.StringType {
-			return diceerrors.NewErrWithFormattedMessage(diceerrors.JSONPathValueTypeErr)
+			return diceerrors.NewErrPathTypeErr(utils.GetJSONFieldType(jsonData))
```

Comment:
nit: can avoid extra code and a function call
```suggestion
			return diceerrors.NewErrWithFormattedMessage(JSONPathValueTypeErr, strings.ToLower(utils.GetJSONFieldType(jsonData)))
```